src/gui/editlocallyjob.h

```diff
@@ -50,6 +51,14 @@ public slots:
     void startSetup();
     void startEditLocally();
 
+private:
+    void fetchRemoteFileParentInfo();
+    bool checkIfFileParentSyncIsNeeded(); // returns true if sync will be needed, false otherwise
+    void startSyncBeforeOpening();
+    void eraseBlacklistRecordForItem();
+    [[nodiscard]] const QString getRelativePathToRemoteRootForFile() const;
+    [[nodiscard]] const QString getRelativePathParent() const;
+
```

Comment:
I would combine this private section with the private section further down the class declaration, splitting these up makes the header more difficult to read